### PR TITLE
docs: release notes for 1.7.0

### PR DESCRIPTION
**Pull request by tigarmo**
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (tigarmo) [inline docs/release-notes/rockcraft-1-7-0.rst:1]: https://canonical-ubuntu-documentation-library--776.com.readthedocs.build/rockcraft/en/776/release-notes/rockcraft-1-7-0/

File: docs/release-notes/index.rst
Comment (tigarmo) [inline docs/release-notes/index.rst:63]: Without this entry, the build fails because the document isn't added in any table of contents. However, *with* this entry this release-notes doc ends up having two listings for a given version:

image


Should this toctree be `:hidden:`?

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (tigarmo) [inline docs/release-notes/rockcraft-1-7-0.rst:3]: @medubelko as we discussed earlier, I ended up putting everything that I would put on a changelog here. Please take a look and point out the items you feel are out-of-place here.

Comment (tigarmo) [inline docs/release-notes/rockcraft-1-7-0.rst:3]: I also removed some sections, like the "Requirements" one, which would be empty for this particular release.

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (tigarmo) [inline docs/release-notes/rockcraft-1-7-0.rst:80]:
```suggestion
The following issues have been resolved in Rockcraft 1.7.0:
```

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (tigarmo) [inline docs/release-notes/rockcraft-1-7-0.rst:93]: the spellchecker is complaining about all of these, heh

File: docs/release-notes/index.rst
Comment (medubelko) [reply docs/release-notes/index.rst:63]: > Should this toctree be :hidden:?

Absolutely.

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (medubelko) [reply docs/release-notes/rockcraft-1-7-0.rst:3]: It was decided that the _Requirements_ section should be mandatory for each page, so it will have to stay in. It's in effect boilerplate.

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (medubelko) [reply docs/release-notes/rockcraft-1-7-0.rst:93]: Annoying. In the short-term, what would be easier?

- Use :literalref: instead
- Add a custom block directive that's a complement for the inline `:spellexception:`

Comment (medubelko) [inline docs/release-notes/rockcraft-1-7-0.rst:93]: In terms of visual presentation, I'm not sure I like splitting the names across lines. I'd prefer one big paragraph.

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (medubelko) [inline docs/release-notes/rockcraft-1-7-0.rst:52]:
```suggestion
A new :ref:`tutorial <build-a-rock-for-a-go-application>` and :ref:`reference
<go-framework-reference>` are available for the go-framework extension.
```

**Review by medubelko — CHANGES_REQUESTED** [review]
This looks good! I did a deeper review of this, because I want to ensure we're off on the right foot.

I agree with your selection of major and minor changes. We'll have better examples to provide to other projects once we have more releases to document.

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (tigarmo) [reply docs/release-notes/rockcraft-1-7-0.rst:93]: It's rendered as a single paragraph; they are split over sequential lines here because of the line-length limit, which basically enforces "one per line" in this case
image

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (tigarmo) [reply docs/release-notes/rockcraft-1-7-0.rst:3]: Alright, but in this case we should review the content - the default template lists "Python 3.x or higher", but in the case of the applications the "official" means of distribution is the snap. In this case, there's no requirement on the user even having Python, but they do need a snap-capable system. Shall I draft a list of requirements here that we can possibly use for the other applications too?

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (tigarmo) [reply docs/release-notes/rockcraft-1-7-0.rst:35]: I don't know where I got this from but I thought we preferred quotes over the inline-code markers. Guess I was wrong!

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (tigarmo) [inline docs/release-notes/rockcraft-1-7-0.rst:39]: maybe this was my confusion: should these be `"ubuntu@22.04"` or inline-code-formatted?

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (tigarmo) [reply docs/release-notes/rockcraft-1-7-0.rst:74]: I had something similar to this originally, but the template points to using the past tense. I can see how it could depend on the specific instance, but won't the notes look awkward if some are in the past tense and some use the present one?

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (tigarmo) [reply docs/release-notes/rockcraft-1-7-0.rst:82]: Sure - I think we need to make it clear-er in the template that these need to be links

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (tigarmo) [reply docs/release-notes/rockcraft-1-7-0.rst]: updated to use `:literalref:`

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (tigarmo) [reply docs/release-notes/rockcraft-1-7-0.rst]: Ah I see. I have a task to document the feature on craft-application, but it won't be ready & integrated into Rockcraft by the end of this week. So I added a call to run ``rockcraft remote-build --help``, which has more information and hopefully the user can take it from there.

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (tigarmo) [reply docs/release-notes/rockcraft-1-7-0.rst]: I updated this paragraph to try to address both concerns (this and the agentless passive voice) - please take a look

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (medubelko) [reply docs/release-notes/rockcraft-1-7-0.rst]: Ah! I reacted instinctively to the code and not the preview, oops.

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (medubelko) [reply docs/release-notes/rockcraft-1-7-0.rst]: Not from me. :hear_no_evil:

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (medubelko) [reply docs/release-notes/rockcraft-1-7-0.rst]: Perfect!

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (medubelko) [reply docs/release-notes/rockcraft-1-7-0.rst]: One more thought about inline code, while I'm thinking about this:

Sometimes a symbol can become a real word. For example, core22 and commands like grep, which were originally intended as machine expressions, have entered general parlance. Things like that, which have become actual words everyday speech, don't need code fences. Determining whether something is a word can sometimes be difficult, so you can leave it up to me.

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (medubelko) [reply docs/release-notes/rockcraft-1-7-0.rst:3]: Yes, please! Feel free to crib from Snapcraft's system reqs, too: https://canonical-snapcraft.readthedocs-hosted.com/en/latest/reference/system-requirements/

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (medubelko) [reply docs/release-notes/rockcraft-1-7-0.rst]: The new copy makes much more sense, thanks!

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (medubelko) [inline docs/release-notes/rockcraft-1-7-0.rst:76]: I'm still not a fan of starting a sentence with a past tense verb, so let's insert ourselves:

```suggestion
We changed the location of the Java runtime executable on rocks built using the
the Maven and Ant plugins from ``/bin/java`` to ``/usr/bin/java``, as the
former conflicts with the user-merged Ubuntu base.
```

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (tigarmo) [reply docs/release-notes/rockcraft-1-7-0.rst:3]: done! Thanks for the cheatsheet

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (tigarmo) [reply docs/release-notes/rockcraft-1-7-0.rst]: I accepted this suggestion "manually", as I worked on the page here

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (tigarmo) [inline docs/release-notes/rockcraft-1-7-0.rst:98]: This entry is new, to reflect the PR that got merged yesterday

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (tigarmo) [inline docs/release-notes/rockcraft-1-7-0.rst:61]: Adding a note to myself here that once #778 is merged I can update these to point to the new docs

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (medubelko) [inline docs/release-notes/rockcraft-1-7-0.rst:101]: > 12-factor

Should we call them "12-factor extensions"? "Django and Flask extensions, for supporting 12-factor apps" might be clearer.

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (medubelko) [inline docs/release-notes/rockcraft-1-7-0.rst:106]:
```suggestion
12-factor app documentation
~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: docs/release-notes/index.rst
Comment (medubelko) [inline docs/release-notes/index.rst:64]: Forgot this bit yesterday. This pattern will keep the TOC cleaner.

```suggestion
   Rockcraft 1.7.0 <rockcraft-1-7-0>
```

**Review by medubelko — CHANGES_REQUESTED** [review]
Three small things, then I think we're good to go.

**Review by medubelko — APPROVED** [review]
:rock: on. Many thanks for being the test subject.

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (tigarmo) [inline docs/release-notes/rockcraft-1-7-0.rst:56]: @medubelko here's what I finally ended up with after much debating with myself

File: docs/release-notes/rockcraft-1-7-0.rst
Comment (tigarmo) [reply docs/release-notes/rockcraft-1-7-0.rst]: reworded this: kept as Django and Flask, and updated the title to `12-factor app extensions`

**Review by medubelko — APPROVED** [review]
Double-approving.

**Review by bepri — APPROVED** [review]
These look great - I'm a big fan of this format!

**Pull request merged**
